pkg/operator/controller/nodeobservability/daemonset.go

```diff
@@ -34,31 +34,29 @@ const (
 // ensureDaemonSet ensures that the daemonset exists
 // Returns a Boolean value indicating whether it exists, a pointer to the
 // daemonset and an error when relevant
-func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeObs *v1alpha2.NodeObservability, sa *corev1.ServiceAccount, ns string) (*appsv1.DaemonSet, error) {
-	nameSpace := types.NamespacedName{Namespace: ns, Name: daemonSetName}
-	desired := r.desiredDaemonSet(nodeObs, sa, ns)
-	if err := controllerutil.SetControllerReference(nodeObs, desired, r.Scheme); err != nil {
-		return nil, fmt.Errorf("failed to set the controller reference for daemonset: %w", err)
+func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeObs *v1alpha2.NodeObservability, sa *corev1.ServiceAccount, ns string, kubeletCAConfigMap *corev1.ConfigMap) (*appsv1.DaemonSet, error) {
+	var (
+		err     error
+		desired *appsv1.DaemonSet
+	)
+	if kubeletCAConfigMap != nil {
+		desired = r.desiredDaemonSet(nodeObs, sa, ns, kubeletCAConfigMap.Name)
+		if err := controllerutil.SetControllerReference(nodeObs, desired, r.Scheme); err != nil {
+			return nil, fmt.Errorf("failed to set the controller reference for daemonset: %w", err)
+		}
 	}
```

Comment:
```suggestion
	desired = r.desiredDaemonSet(nodeObs, sa, ns, kubeletCAConfigMap.Name)
	if err := controllerutil.SetControllerReference(nodeObs, desired, r.Scheme); err != nil {
		return nil, fmt.Errorf("failed to set the controller reference for daemonset: %w", err)
	}
```
We never pass empty kubelet configmap, it doesn't make sense to because we need it for the daemonset, w/o it the daemonset cannot be created.